lib/libc/gen/mips/longjmp.c

```diff
@@ -68,28 +67,29 @@ void longjmp(jmp_buf env, int val) {
   uc.uc_mcontext.__gregs[_REG_V0] = val;
 
   /* Copy saved registers */
-  uc.uc_mcontext.__gregs[_REG_S0] = sc->sc_regs[_R_S0];
-  uc.uc_mcontext.__gregs[_REG_S1] = sc->sc_regs[_R_S1];
-  uc.uc_mcontext.__gregs[_REG_S2] = sc->sc_regs[_R_S2];
-  uc.uc_mcontext.__gregs[_REG_S3] = sc->sc_regs[_R_S3];
-  uc.uc_mcontext.__gregs[_REG_S4] = sc->sc_regs[_R_S4];
-  uc.uc_mcontext.__gregs[_REG_S5] = sc->sc_regs[_R_S5];
-  uc.uc_mcontext.__gregs[_REG_S6] = sc->sc_regs[_R_S6];
-  uc.uc_mcontext.__gregs[_REG_S7] = sc->sc_regs[_R_S7];
-  uc.uc_mcontext.__gregs[_REG_S8] = sc->sc_regs[_R_S8];
+  uc.uc_mcontext.__gregs[_REG_S0] = sc_uc->uc_mcontext.__gregs[_REG_S0];
```

Comment:
Can we define nice macro to hide nicely access to `__gregs`?

Reply:
done